test/gtest/uct/ib/test_ib.cc

```diff
@@ -333,6 +333,59 @@ UCS_TEST_P(test_uct_ib, address_pack) {
 UCT_INSTANTIATE_IB_TEST_CASE(test_uct_ib);
 
 
+class test_uct_ib_utils : public ucs::test {
+};
+
+UCS_TEST_F(test_uct_ib_utils, sec_to_qp_time) {
+    uint8_t qp_val;
+
+    qp_val = uct_ib_to_qp_fabric_time(0);
+    EXPECT_EQ(1, qp_val);
+
+    for (uint8_t i = 1; i < 32; i++) {
+        qp_val = uct_ib_to_qp_fabric_time(4.096 * pow(2, i) / UCS_USEC_PER_SEC);
+        EXPECT_EQ(i % 31, qp_val);
+    }
+
+    qp_val = uct_ib_to_qp_fabric_time(10000);
+    EXPECT_EQ(0, qp_val);
+}
+
+UCS_TEST_F(test_uct_ib_utils, sec_to_rnr_time) {
+    double avg;
+    uint8_t rnr_val;
+
+    rnr_val = uct_ib_to_rnr_fabric_time(0);
+    EXPECT_EQ(1, rnr_val);
+
+    rnr_val = uct_ib_to_rnr_fabric_time(uct_ib_qp_rnr_time_ms[0] / UCS_MSEC_PER_SEC);
+    EXPECT_EQ(1, rnr_val);
+
+    avg = uct_ib_qp_rnr_time_ms[0] * 0.5;
+    rnr_val = uct_ib_to_rnr_fabric_time(avg / UCS_MSEC_PER_SEC);
+    EXPECT_EQ(1, rnr_val);
+
+    for (uint8_t i = 1; i < 2; i++) {
+        avg = (uct_ib_qp_rnr_time_ms[i - 1] + uct_ib_qp_rnr_time_ms[i]) * 0.5;
+
+        rnr_val = uct_ib_to_rnr_fabric_time(uct_ib_qp_rnr_time_ms[i] / UCS_MSEC_PER_SEC);
+        EXPECT_EQ((i + 1) % 32, rnr_val);
+
+        rnr_val = uct_ib_to_rnr_fabric_time(avg / UCS_MSEC_PER_SEC);
+        EXPECT_EQ((i + 1) % 32, rnr_val);
+
+        rnr_val = uct_ib_to_rnr_fabric_time((uct_ib_qp_rnr_time_ms[i - 1] + avg) * 0.5 / UCS_MSEC_PER_SEC);
```

Comment:
isn't middle value checked above?

Reply:
added the comments into the test code to clarify the purposes of the checks